Mathlib/LinearAlgebra/PiTensorProduct.lean

```diff
@@ -613,6 +731,33 @@ theorem reindex_refl : reindex R s (Equiv.refl ι) = LinearEquiv.refl R _ := by
   congr
 #align pi_tensor_product.reindex_refl PiTensorProduct.reindex_refl
 
+variable {t : ι → Type*}
+variable [∀ i, AddCommMonoid (t i)] [∀ i, Module R (t i)]
+
+/-- Re-indexing the components of the tensor product by an equivalence `e` is compatible
+with `PiTensorProduct.map`.-/
+theorem map_comp_reindex_eq (f : Π i, s i →ₗ[R] t i) (e : ι ≃ ι₂) :
+    map (fun i ↦ f (e.symm i)) ∘ₗ reindex R s e = reindex R t e ∘ₗ map f := by
+  ext m
+  simp only [LinearMap.compMultilinearMap_apply, LinearMap.coe_comp, LinearEquiv.coe_coe,
+    LinearMap.comp_apply, reindex_tprod, map_tprod]
+
+theorem map_reindex (f : Π i, s i →ₗ[R] t i) (e : ι ≃ ι₂) (x : ⨂[R] i, s i) :
+    map (fun i ↦ f (e.symm i)) (reindex R s e x) = reindex R t e (map f x) :=
+  DFunLike.congr_fun (map_comp_reindex_eq _ _) _
+
+theorem map_comp_reindex_symm (f : Π i, s i →ₗ[R] t i) (e : ι ≃ ι₂) :
+    map f ∘ₗ (reindex R s e).symm = (reindex R t e).symm ∘ₗ map (fun i => f (e.symm i)) := by
+  ext m
+  apply LinearEquiv.injective (reindex R t e)
+  simp only [LinearMap.compMultilinearMap_apply, LinearMap.coe_comp, LinearEquiv.coe_coe,
+    comp_apply, ← map_reindex, LinearEquiv.apply_symm_apply, map_tprod]
+
+@[simp]
+theorem map_reindex_symm (f : Π i, s i →ₗ[R] t i) (e : ι ≃ ι₂) (x : ⨂[R] i, s (e.symm i)) :
+    map f ((reindex R s e).symm x) = (reindex R t e).symm (map (fun i ↦ f (e.symm i)) x) :=
```

Comment:
It's not clear to me that `simp` is appropriate here, especially as you didn't put it on `map_reindex`. Was there a particular argument you had in mind for doing so?

Reply:
No, I originally put it on buth, then deleted it on `map_reindex` when I noticed it wasn't on `TensorProduct.map_comm`. Of course, there's a very good reason it's not a `TensorProduct.map_comm` and it doesn't apply here, but anyway I'll just go delete the second `simp`.